source-mysql/main.go

```diff
@@ -245,22 +258,57 @@ func (db *mysqlDatabase) connect(ctx context.Context) error {
 	return nil
 }
 
-func (db *mysqlDatabase) getBinlogExpiry() (time.Duration, error) {
+var timeZoneOffsetRegex = regexp.MustCompile(`^[-+][0-9]{1,2}:[0-9]{2}$`)
+
+func queryTimeZone(conn *client.Conn) (*time.Location, error) {
+	var tzName, err = queryStringVariable(conn, `SELECT @@GLOBAL.time_zone;`)
+	if err != nil {
+		return nil, fmt.Errorf("error querying 'time_zone' system variable: %w", err)
+	}
+	logrus.WithField("time_zone", tzName).Debug("queried time_zone system variable")
+	if tzName == "SYSTEM" {
+		return nil, errDatabaseTimezoneUnknown
+	}
+
+	// If it looks like a numeric offset then construct a fixed-offset time zone from that.
+	if timeZoneOffsetRegex.MatchString(tzName) {
+		// The regex requires an explicit +/- sign
+		var sign = 1
+		if tzName[0:1] == "-" {
+			sign = -1
+		}
+		var parts = strings.SplitN(tzName[1:], ":", 2)
+		hours, err := strconv.Atoi(parts[0])
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %q: %w", tzName, err)
+		}
+		minutes, err := strconv.Atoi(parts[1])
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %q: %w", tzName, err)
+		}
+		return time.FixedZone("UTC"+tzName, sign*(hours*60*60+minutes*60)), nil
+	}
+
+	// Otherwise let's hope the time zone setting is a valid IANA zone name.
+	return time.LoadLocation(tzName)
```

Comment:
I kinda want to have an easter egg that says "Thank You" if the timezone is set to UTC 😂